packages/test/test-end-to-end-tests/src/test/sharedInterval.spec.ts

```diff
@@ -571,332 +570,309 @@ describeCompat("SharedInterval", "NoCompat", (getTestObjectProvider, apis) => {
+					interval2,
+					"Unique interval not added",
+				);
+			});
 
-			await provider.ensureSynchronized();
+			it("conflicting removes are visible on both clients", async () => {
+				const { intervals1, intervals2 } = await setupConflictingOps();
+				const interval1 = intervals1.add({ start: 0, end: 0 });
+				const id1 = interval1.getIntervalId();
+				const interval2 = intervals2.add({ start: 0, end: 0 });
+				const id2 = interval2.getIntervalId();
+				await provider.ensureSynchronized();
+				intervals1.removeIntervalById(id2);
+				intervals2.removeIntervalById(id1);
+				await provider.ensureSynchronized();
+				assert.strictEqual(
+					intervals1.getIntervalById(id1),
+					undefined,
+					"Interval not removed from other client",
+				);
+				assert.strictEqual(
+					intervals2.getIntervalById(id2),
+					undefined,
+					"Interval not removed from other client",
+				);
+			});
 
-			assert.strictEqual(
-				interval1,
-				intervals1.getIntervalById(id1),
-				"Interval missing from collection 1",
-			);
-			for (const interval of intervals1) {
-				assert.strictEqual(interval, interval1, "Oddball interval found in client 1");
-			}
+			it("conflicting removes + add resolves to single interval", async () => {
+				const { intervals1, intervals2 } = await setupConflictingOps();
+				const interval1 = intervals1.add({ start: 1, end: 1 });
+				let id1 = interval1.getIntervalId();
+				const interval2 = intervals2.add({ start: 1, end: 1 });
+				const id2 = interval2.getIntervalId();
+				await provider.ensureSynchronized();
+				intervals2.removeIntervalById(id1);
+				intervals1.removeIntervalById(id2);
+				const newInterval1 = intervals1.add({ start: 1, end: 1 });
+				id1 = newInterval1.getIntervalId();
+				await provider.ensureSynchronized();
+				assert.strictEqual(
+					newInterval1,
+					intervals1.getIntervalById(id1),
+					"Interval missing from collection 1",
+				);
+				for (const interval of intervals1) {
+					assert.strictEqual(interval, newInterval1, "Oddball interval found in client 1");
+				}
+				const foundInterval2 = intervals2.getIntervalById(id1);
+				assert.notStrictEqual(foundInterval2, undefined, "Interval missing from collection 2");
+				for (const interval of intervals2) {
+					assert.strictEqual(interval, foundInterval2, "Oddball interval found in client 2");
+				}
+			});
 
-			interval2 = intervals2.getIntervalById(id1);
-			assert.notStrictEqual(interval2, undefined, "Interval missing from collection 2");
-			for (const interval of intervals2) {
-				assert.strictEqual(interval, interval2, "Oddball interval found in client 2");
-			}
+			it("conflicting changes are resolved consistently", async () => {
+				const { intervals1, intervals2 } = await setupConflictingOps();
+				const id1 = intervals1.add({ start: 1, end: 1 }).getIntervalId();
+				await provider.ensureSynchronized();
 
-			if (typeof intervals1.change === "function" && typeof intervals2.change === "function") {
```

Comment:
iirc this is around because the compat matrix extended beyond API changes to `change` (it used to be called something else I guess?) and I didn't particularly care to maintain code that worked the old way.

anyway, if you run into weird errors here that's probably why and you might need to re-add checks like this. IMO this test is kind of weird to have as a compat test anyway.